apireader.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+
+import sys
+import os.path
+import requests
+import json
+from argparse import ArgumentParser
+
+DUMP_NAME = "api_data.dump"
+DIRECTORY_URI = "https://raw.githubusercontent.com/freifunk/directory.api.freifunk.net/master/directory.json"
+
+
+def get_api_dict():
+    response = requests.get(DIRECTORY_URI)
+    api_dict = json.loads(response.text)
+    return api_dict
+
+
+def get_api_data_from_git(api_dict):
+    data = dict()
+    for community in api_dict:
+        api_url = api_dict[community]
+        print("Loading {:<20} {} ...".format(community, api_url), file=sys.stderr)
+        try:
+            response = requests.get(api_url, timeout=2)
+            community_api = response.json()
+            data[community] = community_api
+        except requests.exceptions.RequestException:
+            print("Error: Couldn't load api file for {} from {}.".format(community, api_url), file=sys.stderr)
+            continue
+        except ValueError:
+            print("Error: Couldn't parse api file for {} from {}.".format(community, api_url), file=sys.stderr)
+            continue
+
+    return data
+
+
+def get_api_data_from_file(filename):
+    with open(filename, 'r') as dump_file:
+        api_data = json.load(dump_file)
+    return api_data
+
+
+def get_api_data(api_dict):
+    # NOTE:Unexpected behavior!!
```

Comment:
Reading from "api_data.dump" instead of loading the actual data is not documented.